test/cljam/io/cram/encode/subst_matrix_test.clj

```diff
@@ -0,0 +1,27 @@
+(ns cljam.io.cram.encode.subst-matrix-test
+  (:require [cljam.io.cram.encode.subst-matrix :as subst-mat]
+            [clojure.test :refer [deftest is]]))
+
+(deftest make-subst-matrix-builder-test
+  (let [builder (subst-mat/make-subst-matrix-builder)
+        a->t (subst-mat/assign-code! builder (int \A) (int \T))
+        a->c (subst-mat/assign-code! builder (int \A) (int \C))
+        a->g (subst-mat/assign-code! builder (int \A) (int \G))
+        a->t' (subst-mat/assign-code! builder (int \A) (int \T))
+        g->c (subst-mat/assign-code! builder (int \G) (int \C))
+        g->t (subst-mat/assign-code! builder (int \G) (int \T))
+        t->n (subst-mat/assign-code! builder (int \T) (int \N))
+        n->c (subst-mat/assign-code! builder (int \N) (int \C))]
+    (is (= {\A {\C 1, \G 2, \T 0, \N 3}
```

Comment:
I thought it might be a bit tricky how `MutableInt` starts as a counter and then turns into a code table after the `build-subst-matrix` invocation, but it's more efficient than splitting them into separate objects, so I think it's fine.
It would be helpful to better describe the behavior if we add test cases for counters like `(is (= 2 @a->t @a->t')) ...` before calling `build-subst-matrix`.

Reply:
Thank you for the comment! I totally agree.
Just force-pushed the updated commit which added some tests and comments describing how `MutableInt`s are used and updated.

Reply:
Thank you! With the added comments and test cases, I find the behavior much easier to understand now👍👍